### PR TITLE
Async git

### PR DESCRIPTION
**Pull request by joshaber**
:warning: Dependent on https://github.com/atom/atom/pull/9213 :warning:

Use the new async git API.

### CODE REVIEW

**Comment by joshaber** [comment]
I believe this is ready, but tests won't pass until there's a release of Atom including the async repo work.

File: lib/git-view.coffee [hunk 1]
```diff
@@ -149,18 +164,37 @@ class GitView extends HTMLElement
...
+    @gitStatus.style.display = ''
+
+    Promise.resolve()
+
+  updateStatusText: (repo) ->
+    itemPath = @getActiveItemPath()
+    return unless itemPath
+
+    @updateStatusPromise = @updateStatusPromise
+      .then (_) -> repo?.getCachedPathStatus(itemPath)
```
Comment (kevinsawicki) [inline lib/git-view.coffee]: I think you can :fire: the `(_)` :coffee:

File: lib/git-view.coffee [hunk 2]
```diff
@@ -149,18 +164,37 @@ class GitView extends HTMLElement
...
+      .then (status) =>
+        status = status or 0
```
Comment (kevinsawicki) [inline lib/git-view.coffee]: This might look a bit better as a parameter default:

```coffee
.then (status=0) =>
```

File: lib/git-view.coffee [hunk 3]
```diff
@@ -149,18 +164,37 @@ class GitView extends HTMLElement
...
+
+        if repo?.isStatusNew(status)
+          return @updateAsNewFile()
+
+        if repo?.isStatusModified(status)
+          return @updateAsModifiedFile(repo, itemPath)
+
+        return repo?.isPathIgnored(itemPath).then (ignored) =>
```
Comment (kevinsawicki) [inline lib/git-view.coffee]: Super minor, but I don't think you need the `return` here.

**Comment by joshaber** [comment]
:boom:

**Pull request merged**